src/java/org/apache/cassandra/index/sai/SSTableIndex.java

```diff
@@ -73,13 +80,24 @@ public SSTableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIn
     {
         assert perIndexComponents.context().getValidator() != null;
         this.perIndexComponents = perIndexComponents;
-        this.searchableIndex = perIndexComponents.version().onDiskFormat().newSearchableIndex(sstableContext, perIndexComponents);
+        this.searchableIndex = createSearchableIndex(sstableContext, perIndexComponents);
 
         this.sstableContext = sstableContext.sharedCopy(); // this line must not be before any code that may throw
         this.indexContext = perIndexComponents.context();
         this.sstable = sstableContext.sstable;
     }
 
+    private static SearchableIndex createSearchableIndex(SSTableContext sstableContext, IndexComponents.ForRead perIndexComponents)
+    {
+        if (CassandraRelevantProperties.SAI_INDEX_READS_DISABLED.getBoolean())
```

Comment:
Do we actually need to keep this if we're changing index queriability? The least amount of changes the better I guess?

Reply:
We do need this, that's the actual part we care about: this `SearchableIndex` is created at sstable opening time, not query time, and this is the part that allocate quite a bit with vector, leading to OOM.

Reply:
Ah, of course, pardon my lapse of reason.